library/src/main/cpp/decoder_jxl.cpp

```diff
@@ -73,71 +96,34 @@ std::vector<uint8_t> decodeRGB(const uint8_t* data, size_t size,
       break;
     }
   }
-
-  return pixels;
 }
 
 ImageInfo JpegxlDecoder::parseInfo() {
-  Rect bounds;
-  JxlBasicInfo jxl_info;
+  decode();
 
+  Rect bounds;
   if (cropBorders) {
-    std::vector<uint8_t> pixels =
-        decodeRGB(stream->bytes, stream->size, &jxl_info);
-
-    uint8_t* pixels_buffer = (uint8_t*)pixels.data();
-    uint32_t num_channels =
-        (jxl_info.alpha_bits > 0 ? 1 : 0) + jxl_info.num_color_channels;
-
-    // Re-align gray image to the front of the vector
-    if (num_channels == 2) {
```

Comment:
When would it have even been 2 channels, greyscale with alpha?

Reply:
Yes.